Fix ConfigurationException in integration tests

Bouncheck opened the pull request:
CASSANDRA-15234 introduced several changes to cassandra.yaml
This results in ConifugrationExceptions during ccm start.
As a workaround this change adds as jvmArg "-Dcassandra.allow_new_old_config_keys=true"
if the Cassandra version is at least 4.1.0 and this arg was not added yet.

Additionally any config key explicitly added by driver code has been
conditionally changed to new form depending on Cassandra version.

JvmArg "-Dcassandra.allow_duplicate_config_keys=false" has been added to
ensure that nothing gets implicitly overwritten with wrong value.

avelanarius commented:
Interesting that this issue is still not fixed in the upstream DataStax Java Driver 4.x (they have the same problem even in "master").

avelanarius commented on test-infra/src/main/java/com/datastax/oss/driver/api/testinfra/ccm/CcmBridge.java:
```diff
@@ -324,7 +324,11 @@ public void create() {
       if (getCassandraVersion().compareTo(Version.V2_2_0) >= 0 && !SCYLLA_ENABLEMENT) {
         // @IntegrationTestDisabledScyllaJVMArgs @IntegrationTestDisabledScyllaUDF
-        updateConfArguments.append("enable_user_defined_functions:true").append(' ');
+        if (getCassandraVersion().compareTo(Objects.requireNonNull(Version.parse("4.1.0"))) >= 0) {
```
Minor nit: `Version.parse("4.1.0")` could be extracted to a constant (like `Version.V2_2_0`). Additionally `Objects.requireNonNull` should not be needed.

Bouncheck replied:
Done. 
Will push a new ver in a minute.

Bouncheck committed:
Fix ConfigurationException in integration tests

CASSANDRA-15234 introduced several changes to cassandra.yaml
This results in ConifugrationExceptions during ccm start.
As a workaround this change adds as jvmArg "-Dcassandra.allow_new_old_config_keys=true"
if the Cassandra version is at least 4.1.0 and this arg was not added yet.

Additionally any config key explicitly added by driver code has been
conditionally changed to new form depending on Cassandra version.

JvmArg "-Dcassandra.allow_duplicate_config_keys=false" has been added to
ensure that nothing gets implicitly overwritten with wrong value.

Bouncheck commented:
I've edited the first comment and commit.
It seems that just changing old names to new names may not be enough, as for example new options may now require proper units along with values.
For now this workaround should make do (at least until next major version).

Bouncheck commented:
I've added a commit with the quick fix for "~rcN" labels. Hopefully 5.2.0-rc0 tests pass now.

Bouncheck committed:
Fix SchemaIT

Table system_views.sstable_tasks has slightly changed,
causing `should_get_virtual_metadata()` to fail.
I've added a String of what Cassandra 4.1 seems to be using.
The difference is `task_id` being `timeuuid` now and having additional
`int` column `sstables`.
